### PR TITLE
fix: add old supply offset functions

### PR DESCRIPTION
**Pull request by PaddyMc**
## Description

Exposing functions to allow Osmosis to access the old supply offsets

Osmosis PR => https://github.com/osmosis-labs/osmosis/pull/8751

### CODE REVIEW

File: x/bank/types/keys.go [hunk 1]
```diff
@@ -33,7 +33,12 @@ var (
 	// ParamsKey is the prefix for x/bank parameters
 	ParamsKey = collections.NewPrefix(5)
 
+	// SupplyOffKey is a Osmosis specific key that handles supply offsets
 	SupplyOffsetKey = collections.NewPrefix(88)
+
+	// SupplyOffKeyOld is a Osmosis specific key that handles supply offsets pre v0.50
+	// TODO: Remove in the v28 upgrade
+	SupplyOffsetKeyOld = []byte{0x88}
```
Comment (czarcas7ic) [inline x/bank/types/keys.go:41]: I just want to verify that you checked NewPrefix 88 does not collide with []byte{0x88}

**Review by czarcas7ic — APPROVED** [review]
ACK, we might want to add this to Osmosis specific changes on changelog

File: x/bank/keeper/supply_offset_old.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+	// Bank invariants and IBC requires to remove zero coins.
+	if offsetAmount.IsZero() {
+		supplyOffsetStore.Delete([]byte(denom))
+	} else {
+		supplyOffsetStore.Set([]byte(denom), intBytes)
+	}
+}
+
+// AddSupplyOffset adjusts the current supply offset of a denom by the inputted offsetAmount using the pre v26 (old) key
```
Comment (mattverse) [inline x/bank/keeper/supply_offset_old.go]:
```suggestion
// AddSupplyOffsetOld adjusts the current supply offset of a denom by the inputted offsetAmount using the pre v26 (old) key
```

**Review by mattverse — APPROVED** [review]
LGTM

File: x/bank/types/keys.go (hunk 1 above)
Comment (PaddyMc) [reply x/bank/types/keys.go:41]: Yeah I did, this is the bug

Comment (czarcas7ic) [reply x/bank/types/keys.go:41]: Ah, just understood, thanks

**Pull request merged**